File: src/net/merayen/elastic/ui/objects/top/views/splashview/SplashView.java
package net.merayen.elastic.ui.objects.top.views.splashview;

import net.merayen.elastic.ui.Draw;
import net.merayen.elastic.ui.objects.top.views.View;

public class SplashView extends View {
	private final SplashViewBar bar = new SplashViewBar();

	@Override
	public void onInit() {
		super.onInit();
		add(bar);
	}

	@Override
	public View cloneView() {
		return new SplashView();
	}

	@Override
	public void onDraw(Draw draw) {
		super.onDraw(draw);
		draw.setColor(255, 255, 100);
		draw.setFont("", 100);
		draw.text("Elastic", 10, 150);

		draw.fillOval(450, (float)(Math.sin(System.currentTimeMillis() / 1000.0) * 400) + 400, 100, 100);
	}

	@Override
	public void onUpdate() {
		super.onUpdate();
		bar.width = width;
	}
}

File: src/net/merayen/elastic/ui/objects/top/views/splashview/SplashView.kt
package net.merayen.elastic.ui.objects.top.views.splashview

import net.merayen.elastic.ui.Draw
import net.merayen.elastic.ui.objects.top.views.View

class SplashView : View() {
	private val bar = SplashViewBar()

	override fun onInit() {
		super.onInit()
		add(bar)
	}

	override fun cloneView(): View {
		return SplashView()
	}

	override fun onDraw(draw: Draw) {
		super.onDraw(draw)
		draw.setColor(255, 255, 100)
		draw.setFont("", 100f)
		draw.text("Elastic", 10f, 150f)

		draw.fillOval(450f, (Math.sin(System.currentTimeMillis() / 1000.0) * 400).toFloat() + 400, 100f, 100f)
	}

	override fun onUpdate() {
		super.onUpdate()
		bar.layoutWidth = getWidth()
	}
}
